### PR TITLE
refactor: Make it possible for keywords to be relative to an offer

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
reduce reliance on instanceKeywordRecord in offer().
use a brandToIssuer table to find Issuers

This is the first step of facet parameters (#718). It passes tests and doesn't change any external behavior, so it could be checked in, but my plan is to use it as the base for a branch containing the work toward Facet parameters.

### CODE REVIEW

File: packages/zoe/src/zoe.js [hunk 1]
```diff
@@ -756,49 +764,62 @@ const makeZoe = (additionalEndowments = {}) => {
...
-                  )
-                  .then(_ => proposal.give[keyword]);
-              }
-              // If any other payments are included, they are ignored.
-              return Promise.resolve(
-                amountMathKeywordRecord[keyword].getEmpty(),
-              );
-            });
+            if (giveKeywords.includes(keyword)) {
+              const issuer = brandToIssuer.get(proposal.give[keyword].brand);
```
Comment (katelynsills) [inline packages/zoe/src/zoe.js:804]: nice!

**Review by katelynsills — CHANGES_REQUESTED** [review]
We shouldn't be passing in userKeywords to cleanProposal. that's causing us to perform a check that is essentially not checking anything. Instead, we should compare against the objective keywords until we've gone fully subjective, in which case, we need to check against what the contract provides for that offer, or something similar.

**Comment by Chris-Hibbert** [comment]
> We shouldn't be passing in userKeywords to cleanProposal. that's causing us to perform a check that is essentially not checking anything.
Yeah, you're right that these changes don't stand on their own. 

I now commit that this won't get merged except as a branch with a complete refactoring for the whole Facet Parameters change. Do you want to review it a little at a time, or wait for it to be a coherent whole?

I'll look at the rest of your comments now, and fold in the ones that aren't just pointing out that this change isn't complete on its own.

**Comment by katelynsills** [comment]
> I now commit that this won't get merged except as a branch with a complete refactoring for the whole Facet Parameters change. Do you want to review it a little at a time, or wait for it to be a coherent whole?
> 
> I'll look at the rest of your comments now, and fold in the ones that aren't just pointing out that this change isn't complete on its own.

Ok, that sounds good. I'm ok waiting for the coherent whole.

**Comment by erights** [comment]
I prefer to wait for the coherent whole as well. Thanks.

**Comment by Chris-Hibbert** [comment]
The relevant parts of this have been merged into the Zoe 0.7.0 branch #1143.

**Pull request closed without merging**